Sources/NIO/SelectorEpoll.swift

```diff
@@ -118,23 +118,23 @@ extension Selector: _SelectorBackendProtocol {
         assert(self.timerFD == -1, "self.timerFD == \(self.timerFD) in deinitAssertions0, forgot close?")
     }
 
-    func register0<S: Selectable>(selectable: S, fd: Int, interested: SelectorEventSet) throws {
+    func register0<S: Selectable>(selectable: S, fd: Int, interested: SelectorEventSet, sequenceIdentifier: RegistrationSequenceIdentifier) throws {
         var ev = Epoll.epoll_event()
         ev.events = interested.epollEventSet
-        ev.data.fd = Int32(fd)
+        ev.data.u64 = UInt64(UInt64(sequenceIdentifier) << 32 + UInt64(fd)) // we put fd in lower 4 bytes to allow eventfd & timerfd to work as is
```

Comment:
Nit: the outer `UInt64` constructor isn't necessary.
```suggestion
        ev.data.u64 = UInt64(sequenceIdentifier) << 32 + UInt64(fd) // we put fd in lower 4 bytes to allow eventfd & timerfd to work as is
```